evm/p2p/discovery.py

```diff
@@ -0,0 +1,357 @@
+import asyncio
+import logging
+import time
+
+import rlp
+from eth_utils import (
+    decode_hex,
+    is_integer,
+    force_bytes,
+)
+
+from evm.ecc import get_ecc_backend
+from evm.p2p import kademlia
+from evm.utils.secp256k1 import private_key_to_public_key
+from evm.utils.keccak import keccak
+from evm.utils.numeric import (
+    big_endian_to_int,
+    int_to_big_endian,
+    safe_ord,
+)
+
+
+logger = logging.getLogger("discovery")
```

Comment:
I can't say I'll enjoy the extra 'self.' typing that will incur, but it might be worth it

Reply:
I'd be fine with a convenience API such as `self.l or self.log` as long as it was a reference to `self.logger` so that it's still really clear what it is when anyone reads the code.